src/routes/admin/_admin.tsx

```diff
@@ -17,6 +30,24 @@ export const Route = createFileRoute('/admin/_admin')({
 function LayoutComponent() {
   const [route, setRoute] = useState<string>('/admin/xpub');
   const { pathname } = useLocation();
+  const queryClient = useQueryClient();
+  const [isRefreshing, setIsRefreshing] = useState(false);
+
+  const onRefreshClick = () => {
+    setIsRefreshing(true);
+    queryClient
+      .invalidateQueries()
+      .then(() => {
+        toast.success('Data refreshed!');
+      })
+      .catch((err) => {
+        toast.error(err.message);
+        logger.error(err);
+      })
+      .finally(() => {
+        setIsRefreshing(false);
+      });
+  };
```

Comment:
Didn't add setTimeout because of "animation shouldn't have a higher priority than data updating" is a general principle in web development and UX design

Reply:
How does this violate the principle you mentioned?
- The awaiting waits only for setting the `isRefreshing` state. 
- The data update is not postponed - the tables are updated instantly - only the RefreshButton's icon is spinning for at least 500 ms - which is good for UX

So I don't get your point...

Reply:
So? Why you haven't applied the minimal timeout which I suggested?

Reply:
Because tanstack query have tools to handle this by itself

Reply:
But the behaviour is completely different.
1. You adds a message at the bottom (while User eyes looks at the RefreshButton which is at the top) 
![yourMethod](https://github.com/user-attachments/assets/b1923a0e-50b9-43ea-a64f-74c8c3a8bad5)
 
_____

2. My suggestion proposes a minimal time for animation for the arrows in RefreshButton
![mySuggestion](https://github.com/user-attachments/assets/010ca120-6830-40f8-ad07-3d6cd46c117e)

Reply:
I agree with @chris-4chain that the version with longer animation then real update is much better from the user experience perspective, because I see that after clicking the button the refresh has happened while in the current version I se almost nothing, and probably as user I will click this button many times because I will think that it's not working.

Reply:
I added info message that show if data is updated or not. Agree with animation for users. That what i got. My suggestion can be to add one time animation instead of infinity. If you prefer option below. I can apply your suggestion

https://github.com/user-attachments/assets/89e8055c-511c-47f9-8c75-f89e84da3af9

Reply:
I like the option presented on your movie.